### ISSUE TITLE
GUI: recovery shows "Installling update" when doing a factory reset

### ISSUE THREAD

**Issue by amartinz** [ubports/halium_bootable_recovery#18]
When issuing a factory reset from the system, it will reboot to recovery to, well, factory reset the device.
To do so, it will write to `/cache/recovery/ubuntu_command` and trigger `/sbin/system-image-upgrader`.

Currently there is no way to check what we are actually doing and even if not installing an OTA and just formatting data, it will show "Installing update".

We should smarten up the `system-image-upgrader` specific code and better represent the UI states.

**Comment by Flohack74** [comment]
And a progress bar please ^^

**Comment by amartinz** [comment]
Will contact you next week regarding mockups :)

**Comment by Flohack74** [comment]
Luckily at least LOS recovery 5.1/7.1/9.0 has a progress bar already included, but we do not use it since our script runs as external process and cannot report back into the C code. I think that can be fixed somehow =)

**Comment by amartinz** [comment]
Yes, it can. We need to define steps to report and add some callbacks to update UI state.

As we can not really report percentages of the update progress i think it would be best to show an indeterminated progress bar and update text according to what is happening?

Then we can reuse the current yumi animation we have and just update the text.

**Comment by Flohack74** [comment]
Yeah the old 5.1 recovery uses "indeterminate" mode already, below Yumi there are 3 orange dots moving around... But I think we lost that over time and the many recovery changes LOS has done. And yes, a text update facility would be great
